tests/backup/backup_rbac_test.go

```diff
@@ -499,3 +499,433 @@ var _ = Describe("{VerifyRBACforInfraAdmin}", func() {
+		postRuleNameMap               = make(map[string]string)
+		preRuleUidMap                 = make(map[string]string)
+		postRuleUidMap                = make(map[string]string)
+		numOfUsers                    = 3
+		appAdminUser                  string
+		adminCredName                 string
+		adminCloudCredUID             string
+		adminBackupLocationName       string
+		adminBackupLocationUID        string
+		appAdminBackupLocationName    string
+		appAdminBackupLocationUID     string
+		customRoleName                backup.PxBackupRole
+		userClusterMap                = make(map[string]map[string]string)
+		backupNameMap                 = make(map[string]string)
+		restoreNameMap                = make(map[string]string)
+		userBackupNamesMap            = make(map[string][]string)
+		backupLocationMap             = make(map[string]string)
+		backupLocationsFromAdmin      []string
+		userNames                     = make([]string, 0)
+		providers                     = getProviders()
+		bkpNamespaces                 []string
+		appAdminRole                  backup.PxBackupRole = backup.ApplicationOwner
+		labelSelectors                map[string]string
+		mutex                         sync.Mutex
+	)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("VerifyRBACforAppAdmin", "Validates the RBAC operation for app-admin user.", nil, 87888)
+		log.InfoD("scheduling applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, appCtx := range appContexts {
+				appCtx.ReadinessTimeout = appReadinessTimeout
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+				namespace := GetAppNamespace(appCtx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+
+	It("Validates the RBAC operation for app-admin user", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step(fmt.Sprintf("Create a user with %s role", appAdminRole), func() {
+			log.InfoD(fmt.Sprintf("Creating a user with %s role", appAdminRole))
+			appAdminUser = createUsers(1)[0]
+			err := backup.AddRoleToUser(appAdminUser, appAdminRole, fmt.Sprintf("Adding %v role to %s", appAdminRole, appAdminUser))
+			log.FailOnError(err, "failed to add role %s to the user %s", appAdminRole, appAdminUser)
+		})
+
+		Step("Verify App-Admin User doesnt have permission to create cloud credential", func() {
+			log.InfoD("Verify App-Admin User doesnt have permission to create cloud credential")
+			nonAdminCtx, err := backup.GetNonAdminCtx(appAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appAdminUser)
+			for _, provider := range providers {
+				if provider != drivers.ProviderNfs {
+					cloudCredentialName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
```

Comment:
Can we please change time.Now().Unix() to RandomString() everywhere.
This will prevent cascading issue in case deletion of any object fails for this testcase